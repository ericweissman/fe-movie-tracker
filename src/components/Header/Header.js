import React from 'react'
import PropTypes from "prop-types"
import { NavLink, Link } from 'react-router-dom'

export const Header = ({ logoutUser, user }) => {
  return (
    <header className="header">
      <Link to='/' className='nav'>
        <h1>Movie Tracker</h1>
        </Link>
      <nav>
        <NavLink to="/" className="nav">
          Show All Movies
        </NavLink>
        {!user.id && (
          <NavLink to="/login" className="nav">
            Login
          </NavLink>
        )}
        {!user.id && (
          <NavLink to="/signup" className="nav">
            Signup
          </NavLink>
        )}
        {!user.id && (
          <NavLink to="/login" className="nav">
            View Favorites
          </NavLink>
        )}
        {user.id && (
          <NavLink to="/favorites" className="nav">
            View Favorites
          </NavLink>
        )}
        {user.id && (
          <Link to="/">
            <button onClick={logoutUser}>Log Out</button>
          </Link>
        )}
      </nav>
    </header>
  )
}

Header.propTypes = {
  logoutUser: PropTypes.func,
  user: PropTypes.object
}