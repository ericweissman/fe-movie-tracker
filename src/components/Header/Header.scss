@font-face {
  font-family: 'Cereal';
  src: url(./AirbnbCereal-Black.ttf)
}

.header {
  align-items: center;
  display: flex;
  justify-content: space-between;
  margin: 0 15px;
  font-family: 'Cereal';
  text-transform: uppercase;
  
  h1 {
    color: #1f1f1f;
  }

  button {
    background-image: url(./sign-out-alt-solid.svg);
    background-repeat: no-repeat;
    border: none;
    cursor: pointer;
    height: 30px;
    width: 30px;
  }

  .nav {
    color: #1f1f1f;
    margin: 0 7px;
    text-decoration: none;

    &:hover {
      color: blue;
    }
  }
}