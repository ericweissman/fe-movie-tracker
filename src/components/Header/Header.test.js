import React from 'react'
import { Header } from './Header'
import { shallow } from 'enzyme'

let wrapper;
const logoutUserMock = jest.fn();
const userMock1 = {name: 'Eric', id: 2}
const noUserMock = {}

describe('Header', () => {
  it('should match the correct snapshot when there is a user in the store', () => {
    wrapper = shallow(<Header  logoutUser={logoutUserMock} user={userMock1} />)
    expect(wrapper).toMatchSnapshot()  
  });

  it('should match the correct snapshot when there is not a user in the store', () => {
    wrapper = shallow(<Header logoutUser={logoutUserMock} user={noUserMock} />)
    expect(wrapper).toMatchSnapshot()
  })

})