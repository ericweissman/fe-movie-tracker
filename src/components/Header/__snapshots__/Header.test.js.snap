// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the correct snapshot when there is a user in the store 1`] = `
<header>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="nav"
    to="/"
  >
    Show All Movies
  </NavLink>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="nav"
    to="/login"
  >
    Login
  </NavLink>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="nav"
    to="/signup"
  >
    Signup
  </NavLink>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="nav"
    to="/favorites"
  >
    View Favorites
  </NavLink>
  <button
    onClick={[MockFunction]}
  >
    LOG OUT
  </button>
</header>
`;

exports[`Header should match the correct snapshot when there is not a user in the store 1`] = `
<header>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="nav"
    to="/"
  >
    Show All Movies
  </NavLink>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="nav"
    to="/login"
  >
    Login
  </NavLink>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="nav"
    to="/signup"
  >
    Signup
  </NavLink>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="nav"
    to="/favorites"
  >
    View Favorites
  </NavLink>
</header>
`;
